packages/e2e/test/local-network/local-network.test.ts

```diff
@@ -108,4 +109,190 @@ describe('Local Network', () => {
+          address: sendingAddress,
+          value: {
+            assets: new Map([[assetId, 1n]]),
+            coins: 50_000_000n
+          }
+        }
+      ]),
+      scripts: [script]
+    };
+    const unsignedTx = await wallet.initializeTx(txProps);
+
+    const finalizeProps = {
+      extraSigners: [extraSigner],
+      scripts: [script],
+      tx: unsignedTx
+    };
+
+    const signedTx = await wallet.finalizeTx(finalizeProps);
+    await wallet.submitTx(signedTx);
+
+    // Wait until wallet is aware of the minted token.
+    await firstValueFrom(
+      wallet.balance.utxo.total$.pipe(filter(({ assets }) => (assets ? assets.has(assetId) : false)))
+    );
+
+    // Wait until wallet is aware of the burned token.
+    await firstValueFrom(wallet.balance.utxo.total$);
+
+    // Search chain history to see if the transaction is there.
+    const txFoundInHistory = await firstValueFrom(
+      wallet.transactions.history$
+        .pipe(filter((txs) => txs.filter((tx) => tx.id === signedTx.id).length === 1))
+        .pipe(map((txs) => (txs.length > 0 ? txs.find((tx) => tx.id === signedTx.id) : undefined)))
+    );
+
+    expect(txFoundInHistory).toBeDefined();
+    expect(txFoundInHistory?.id).toEqual(signedTx.id);
+
+    wallet.shutdown();
+  });
+
+  it('will burn an NFT using native scripts.', async () => {
+    const wallet = (await getWallet({ env, idx: 10, name: 'Minting Wallet', polling: { interval: 1000 } })).wallet;
+
+    const minterKeyAgent = await KeyManagement.InMemoryKeyAgent.fromBip39MnemonicWords(
+      {
+        getPassword: async () => Buffer.from(''),
+        mnemonicWords: env.KEY_MANAGEMENT_PARAMS.mnemonic.split(' '),
+        networkId: Cardano.NetworkId.testnet
+      },
+      { inputResolver: { resolveInputAddress: async () => null } }
+    );
+
+    const policyKeyDerivationPath = {
+      index: 0,
+      role: KeyManagement.KeyRole.External
+    };
+
+    const derivedAddress = await minterKeyAgent.deriveAddress({
+      index: 0,
+      type: KeyManagement.AddressType.External
+    });
+
+    await firstValueFrom(wallet.syncStatus.isSettled$.pipe(filter((isSettled) => isSettled)));
+
+    const [{ address: sendingAddress }] = await firstValueFrom(wallet.addresses$);
+
+    const keyHash = Cardano.policyKeyHash(derivedAddress.address);
+    const extraSigner = new KeyManagement.util.KeyAgentTransactionSigner(minterKeyAgent, policyKeyDerivationPath);
+
+    const script: Cardano.NativeScript = {
+      __type: Cardano.ScriptType.Native,
+      kind: Cardano.NativeScriptKind.RequireAllOf,
+      scripts: [
+        {
+          __type: Cardano.ScriptType.Native,
+          keyHash: Cardano.Ed25519KeyHash(keyHash),
+          kind: Cardano.NativeScriptKind.RequireSignature
+        }
+      ]
+    };
+
+    const policyId = Cardano.nativeScriptPolicyId(script);
+    const assetName = Cardano.AssetName('BABABABABABABA');
+    const assetId = Cardano.AssetId(policyId.toString() + assetName.toString());
```

Comment:
I think we should introduce here the higher abstraction **AssetClass** domain concept:

In haskell we have:

``` haskell
newtype AssetClass = AssetClass { unAssetClass :: (CurrencySymbol, TokenName) }
```


So TokenMap, instead of taking the asset id directly, could take an asset class:

``` javascript
let assetClass = {
  tokenName: "XXXX", // This is the asset name
  currencySymbol: "YYYY" // This is the policy ID
}
```
https://github.com/input-output-hk/plutus/blob/4b40b31893a8a31456946670af981b06d71be6d5/plutus-ledger-api/src/PlutusLedgerApi/V1/Value.hs#L138

We could then get the asset ID under the hood to add to the transaction.

@rhyslbw @mkazlauskas

Reply:
Btw using the asset class concept, the lovelace coins should also be part of the token map as:

``` javascript
let assetClass = {
  tokenName: 'lovelace',
  currencySymbol: '' // The lovelace asset currency symbol is an empty string
}
```

Reply:
Noted. Deferring to a future ticket.